python/hail/methods/statgen.py

```diff
@@ -2997,50 +3007,42 @@ def ld_prune(call_expr, r2=0.2, bp_window_size=1000000, memory_per_core=256, blo
         field = Env.get_uid()
         mt = mt.select_entries(**{field: call_expr})
 
-    sites_only_table = _local_ld_prune(require_biallelic(mt, 'ld_prune'), field, r2, bp_window_size, memory_per_core)
-
-    sites_path = new_temp_file()
-    sites_only_table.write(sites_path, overwrite=True)
-    sites_only_table = hl.read_table(sites_path)
-
-    locally_pruned_ds = mt.annotate_rows(pruned=sites_only_table[mt.row_key])
-    locally_pruned_ds = locally_pruned_ds.filter_rows(hl.is_defined(locally_pruned_ds.pruned))
-
-    locally_pruned_ds = (locally_pruned_ds.annotate_rows(
-        mean=locally_pruned_ds.pruned.mean,
-        centered_length_sd_reciprocal=locally_pruned_ds.pruned.centered_length_sd_reciprocal))
-
-    locally_pruned_path = new_temp_file()
-    locally_pruned_ds.write(locally_pruned_path, overwrite=True)
+    locally_pruned_table_path = new_temp_file()
+    (_local_ld_prune(require_biallelic(mt, 'ld_prune'), field, r2, bp_window_size, memory_per_core)
+        .add_index()
+        .write(locally_pruned_table_path, overwrite=True))
+    locally_pruned_table = hl.read_table(locally_pruned_table_path)
 
-    locally_pruned_ds = hl.read_matrix_table(locally_pruned_path)
+    locally_pruned_ds_path = new_temp_file()
+    mt = mt.annotate_rows(
+        mean=locally_pruned_table[mt.row_key].mean,
```

Comment:
I had written this join as `mt.annotate_rows(pruned=sites_only_table[mt.row_key])` instead of adding the `mean` and `centered_length_rec` separately, because I was told each expression that looks like table[key] does its own join. Have you been benchmarking how long the method takes with these changes, versus before?

Reply:
good catch, i'll switch that back